fe/fe-core/src/main/java/org/apache/doris/planner/DistributedPlanner.java

```diff
@@ -60,9 +60,18 @@
     private final static Logger LOG = LogManager.getLogger(DistributedPlanner.class);
 
     private final PlannerContext ctx_;
+    private final long autoBroadcastJoinThreshold;
 
     public DistributedPlanner(PlannerContext ctx) {
         ctx_ = ctx;
+        long perNodeMemLimit = ctx_.getQueryOptions().mem_limit;
```

Comment:
It seems to be more cohesive to put the logic function that calculates autoBroadcastJoinThreshold in 'createHashJoinFragment'.